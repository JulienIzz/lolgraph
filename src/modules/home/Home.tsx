import styled from "@emotion/native";
import { BaseScreenNoBack } from "../../shared/screen/BaseScreenNoBack";
import { HomeButton } from "./HomeButton";
import { useNavigation } from "@react-navigation/native";

export type HomeRouteProps = undefined;

export const Home = () => {
  const navigation = useNavigation();
  return (
    <BaseScreenNoBack title="MyLEAGUE">
      <Container>
        <HomeButton
          label="Profile"
          icon={require("../../../assets/utils/user.png")}
          onPress={() => navigation.navigate("ProfileInput")}
        />
        <HomeButton
          label="Timers"
          icon={require("../../../assets/utils/clock.png")}
          onPress={() => navigation.navigate("ProfileInput")}
        />
      </Container>
    </BaseScreenNoBack>
  );
};

const Container = styled.View({
  flex: 1,
  justifyContent: "center",
  alignItems: "center",
});
