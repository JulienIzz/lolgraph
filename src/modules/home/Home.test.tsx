import { render, screen } from "@testing-library/react-native";
import { NavigationContainer } from "@react-navigation/native";
import { SafeAreaProvider } from "react-native-safe-area-context";
import { Home } from "./Home";

jest.mock(
  "react-native-safe-area-context",
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  () => require("react-native-safe-area-context/jest/mock").default,
);

jest.mock("@react-navigation/native", () => ({
  ...jest.requireActual("@react-navigation/native"),
  useNavigation: () => ({ navigate: jest.fn() }),
}));

test("home screen with title is correctly rendered", async () => {
  const expectedHeaderTextForTest = "MyLEAGUE";
  const expectedProfileButton = "Profile";
  const expectedTimersButton = "Timers";

  render(
    <SafeAreaProvider>
      <NavigationContainer>
        <Home />
      </NavigationContainer>
    </SafeAreaProvider>,
  );

  const HeaderOutput = screen.getByText(expectedHeaderTextForTest);
  const ProfileButtonOutput = screen.getByText(expectedProfileButton);
  const TimersButtonOutput = screen.getByText(expectedTimersButton);

  expect(HeaderOutput).toBeTruthy();
  expect(ProfileButtonOutput).toBeTruthy();
  expect(TimersButtonOutput).toBeTruthy();
  expect(screen.toJSON()).toMatchSnapshot();
});
