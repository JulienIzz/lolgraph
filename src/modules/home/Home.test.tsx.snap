// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home screen with title is correctly rendered 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    insets={
      {
        "bottom": 0,
        "left": 0,
        "right": 0,
        "top": 0,
      }
    }
    style={
      [
        {
          "backgroundColor": "#00B4D8",
          "height": 60,
          "paddingTop": 0,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 20,
              "fontWeight": "bold",
            },
            undefined,
          ]
        }
      >
        MyLEAGUE
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#00B4D8",
          "borderRadius": 6,
          "display": "flex",
          "height": 80,
          "justifyContent": "center",
          "margin": 10,
          "opacity": 1,
          "width": 220,
        }
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "height": 30,
              "justifyContent": "center",
              "left": 20,
              "position": "absolute",
              "width": 30,
            },
            undefined,
          ]
        }
      >
        <Image
          source={1}
          style={
            [
              {
                "height": "100%",
                "resizeMode": "cover",
                "width": "100%",
              },
              undefined,
            ]
          }
        />
      </View>
      <Text
        style={
          [
            {
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Profile
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#00B4D8",
          "borderRadius": 6,
          "display": "flex",
          "height": 80,
          "justifyContent": "center",
          "margin": 10,
          "opacity": 1,
          "width": 220,
        }
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "height": 30,
              "justifyContent": "center",
              "left": 20,
              "position": "absolute",
              "width": 30,
            },
            undefined,
          ]
        }
      >
        <Image
          source={1}
          style={
            [
              {
                "height": "100%",
                "resizeMode": "cover",
                "width": "100%",
              },
              undefined,
            ]
          }
        />
      </View>
      <Text
        style={
          [
            {
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Timers
      </Text>
    </View>
  </View>
</View>
`;
